test/format/config.c

```diff
@@ -331,11 +328,43 @@ config_encryption(void)
 
 /*
  * config_in_memory --
- *	In-memory configuration.
+ *	Periodically set up an in-memory configuration.
  */
 static void
 config_in_memory(void)
 {
+	/*
+	 * Configure in-memory before configuring anything else, in-memory has
+	 * many related requirements. Don't configure in-memory if there's any
+	 * incompatible configurations, so we don't have to configure in-memory
+	 * every time we configure something like LSM, that's too painful.
```

Comment:
The comment mentions LSM, but where do we check for LSM?  The others look good.  I think we want to add "compact" to the list.